src/pykeen/triples/triples_factory.py

```diff
@@ -312,7 +317,46 @@ def restrict_triples(
 
 
 @dataclasses.dataclass
-class CoreTriplesFactory:
+class DatasetInfo:
```

Comment:
Maybe `TriplesFactoryInfo` so this doesn't get confused with the actual `Dataset ` class hierarchy